### DISCUSSION TITLE
What did I wrong?

### DISCUSSION THREAD

**Discussion by gntz555** [swar/Swar-Chia-Plot-Manager#101] [Q&A]
When I start view in powershell I get this error:

PS C:\Swar-Chia-Plot-Manager> python manager.py view
python : Traceback (most recent call last):
In Zeile:1 Zeichen:1
+ python manager.py view
+ ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Traceback (most recent call last)::String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
  File "C:\Swar-Chia-Plot-Manager\manager.py", line 36, in <module>
    view()
  File "C:\Swar-Chia-Plot-Manager\plotmanager\library\utilities\commands.py", line 80, in view
    drive = job.temporary_directory.split('\\')[0]
AttributeError: 'NoneType' object has no attribute 'split'

Here is my "config.yaml" >> [config.txt](https://github.com/swar/Swar-Chia-Plot-Manager/files/6411861/config.txt)

OS: Windows 10 Pro
CPU: AMD Ryzen x5950
RAM: Patriot Viper 32GB 3600 MHz 
Motherboard: ASUS Prime x570 Pro
OS SSD: Seagate FireCuda 520, NVMe PCIe X4, 500 GB
SSD: 3,84TB 2,5 Micron 5300 PRO Enterprise
Plot final directory is on my WD Book external HDD

I really don't know what I should do. I lost my plots with the chia GUI because of wrong settings twice now. I really need help please.

**Comment by Amneamnius** [comment]
You're missing "temporary_directory:" that is not an optional parameter. Also your destination_directory _might_ need quotes due to the spaces "D:\chia final plot directory"

**Comment by gntz555** [comment]
> You're missing "temporary_directory:" that is not an optional parameter. Also your destination_directory _might_ need quotes due to the spaces "D:\chia final plot directory"

I'll try it ASAP. I started 2 plots in the meanwhile with the GUI. Once it's finished I will test it and report back.

**Comment by Amneamnius** [comment]
You should be able to start the manager or restart it with GUI plots running and it should pickup the current plots

**Comment by swar** [comment]
@gntz555 please share your config.yaml.

@Amneamnius  is correct. It shouldn't kill any plots at all

**Comment by gntz555** [comment]
> @gntz555 please share your config.yaml.
> 
> @Amneamnius is correct. It shouldn't kill any plots at all

Oh I didn't know! With my listed hardware, are this the right settings? 
Here is the config with the changes Amneamnius recommended. [config.txt](https://github.com/swar/Swar-Chia-Plot-Manager/files/6412197/config.txt)

**Comment by Amneamnius** [comment]
Those settings might be fine but would be maxed at ssd read/write speed and cpu usage when you have 10 concurrent plots going after they're staggering.
Looks like you're using your main drive as temp2? Just be careful that a lot of usage could end up killing the drive.

**Comment by gntz555** [comment]
> Those settings might be fine but would be maxed at ssd read/write speed and cpu usage when you have 10 concurrent plots going after they're staggering.
> Looks like you're using your main drive as temp2? Just be careful that a lot of usage could end up killing the drive.

Hey thank you a lot! Yes I initially didn't intend to use the C drive as temp, but I figured that the write/ read of my single micron SSD is a big cap. Now I have encountered the problem that as my C drive gets full, it causes the "Error and retry in 5 minutes". slowing down my plotting. I selected here the two SSD as temp in expectation that they will act similar to a Raid 0 system, allowing to utilize both drives effectifly, but this is not the case.
![Bild_2021-05-03_004820](https://user-images.githubusercontent.com/83549410/116830268-4166a200-aba9-11eb-8e8c-73a2c94dfede.png)

**Comment by gntz555** [comment]
![Bild_2021-05-03_004923](https://user-images.githubusercontent.com/83549410/116830307-678c4200-aba9-11eb-8e7f-6c0cea0ebeba.png)

**Comment by Amneamnius** [comment]
With the config you provided its writing to E:\ first, then to C:\ for Phase 3/4 (so this will fill up for sure). Finally it copies from C:\ to D:\
Your C:\ doesn't have enough space for more than 1 plot so you shouldn't use it, I think 1 plot needs about 330GB of temporary space so even your E:\ would be maxed out when it reaches 10 concurrent plots.

I'd say just lower max concurrent to about 5-6 and look at drive usage before increasing it. You would edit the yalm file and do "manager.py restart" for it to add new plots if it can.

**Comment by gntz555** [comment]
Thank you very much! I edited the file to this settings. Does that seem okay?
    threads: 12
    buckets: 128
    memory_buffer: 4800
    max_concurrent: 6
    max_concurrent_with_start_early: 10
    stagger_minutes: 20
    max_for_phase_1: 4
    concurrency_start_early_phase: 4
    concurrency_start_early_phase_delay: 0
    temporary2_destination_sync: false

What I'm not sure about is the RAM. Is this in MB or in MiB?

**Comment by Amneamnius** [comment]
Yeah, you have 32 threads though and more than 4 might not be all that helpful (from what I've read on reddit) so maybe see how that does vs 8 threads.

**Comment by gntz555** [comment]
I tried to start it now, but wtf?

![image](https://user-images.githubusercontent.com/83549410/116832674-48e07800-abb6-11eb-8231-3b9fe1274a43.png)

**Comment by Amneamnius** [comment]
Something in your config with a "C" character. Can you paste it? And if you're making edits you can just use "manager.py restart" rather than stopping/starting each time.

**Comment by gntz555** [comment]
[config.txt](https://github.com/swar/Swar-Chia-Plot-Manager/files/6412509/config.txt)


Alright, thank you again and sorry that I'm bothering you with this.

**Comment by Amneamnius** [comment]
[config (1).txt](https://github.com/swar/Swar-Chia-Plot-Manager/files/6412516/config.1.txt)
Can you try this one, I edited the directory names to not be in quotes and use _ instead of spaces (it wasn't passing a YALM validator). There were also some spaces that might have caused issues.

Also you wanna remove the previous one as it has discord info

**Comment by gntz555** [comment]
[config.txt](https://github.com/swar/Swar-Chia-Plot-Manager/files/6412519/config.txt)

FINALLY! I changed the directory names.

**Comment by gntz555** [comment]
> [config (1).txt](https://github.com/swar/Swar-Chia-Plot-Manager/files/6412516/config.1.txt)
> Can you try this one, I edited the directory names to not be in quotes and use _ instead of spaces (it wasn't passing a YALM validator). There were also some spaces that might have caused issues.
> 
> Also you wanna remove the previous one as it has discord info

We actually had the same idea. Haha :D 

*already swapped the files, with edited discord link

**Comment by gntz555** [comment]
![image](https://user-images.githubusercontent.com/83549410/116833231-57c82a00-abb8-11eb-9448-3c39cc1228ee.png)

**Comment by Amneamnius** [comment]
Nice! it should start adding in new jobs to hit your max for phase1 and then keep going as things are finishing up.

**Comment by gntz555** [comment]
Hey man if I ever make money with this, I won't forget you. WOOOOF what a relief!

**Comment by gntz555** [comment]
Need, or rather can I manually uptade the view, because I see no other jobs.

**Comment by Amneamnius** [comment]
It will update every 60s, you should see the screen flicker since its rewriting stuff. And it should basically have added up to 4 jobs now in phase1

**Comment by gntz555** [comment]
![image](https://user-images.githubusercontent.com/83549410/116833406-3ae02680-abb9-11eb-92cf-f5b8db784866.png)

Still just 1 plot. I also thought it would have 4 jobs.

**Comment by Amneamnius** [comment]
Check the logs for any errors? You can also set  log_level: ERROR to INFO to get more in the logs.

**Comment by gntz555** [comment]
[micron_2021-05-03_02_32_49_845378.log](https://github.com/swar/Swar-Chia-Plot-Manager/files/6412550/micron_2021-05-03_02_32_49_845378.log)
That's the log.

**Comment by gntz555** [comment]
No errors, but is it because I deleted the "inland" part?

**Comment by Amneamnius** [comment]
I think manager.log where the manager.py file is would have any errors/info saved that might be useful here.

No Inland was another job as an example to have multiple temporary drives.

**Comment by gntz555** [comment]
Now after the 20 minutes the second plot started:
![image](https://user-images.githubusercontent.com/83549410/116833711-1127ff00-abbb-11eb-9f7b-a1168d3d9694.png)

**Comment by Amneamnius** [comment]
Ah okay I missed it, you have stagger_minutes: 20 so it'll wait 20m before each new job at the start. Eventually the stagger will just be based on when the plots finish.

**Comment by gntz555** [comment]
There is no such file here. 
![image](https://user-images.githubusercontent.com/83549410/116833771-56e4c780-abbb-11eb-8167-143a9d0d5a48.png)

The manager.txt file is empty.

**Comment by gntz555** [comment]
> Ah okay I missed it, you have stagger_minutes: 20 so it'll wait 20m before each new job at the start. Eventually the stagger will just be based on when the plots finish.

Ah okay, so I thought the base is set with 4 plots, THEN it will stagger every 20 minutes. So should I have left the stagger at 0?

**Comment by Amneamnius** [comment]
Yeah that works, they will naturally stagger from then on as things finish.

**Comment by gntz555** [comment]
So how would I approach this? Just edit the config.YAML and restart the manager. But when can I restart it then?

**Comment by Amneamnius** [comment]
Yes save it and restart. To get the prompt just CTRL+C at the powershell, run "manager.py restart" then "manager.py view"

**Comment by gntz555** [comment]
How can I save it? The valid options are "start", "restart", "stop", "view", and "analyze_logs".

**Comment by Amneamnius** [comment]
Edit it in notepad/notepad++/etc... and save the yalm, then do manager.py restart and it'll load those settings and start using from then on.

**Comment by gntz555** [comment]
Ah okay, I thought I need to save the plot progress via powershell somehow.

![image](https://user-images.githubusercontent.com/83549410/116834197-459cba80-abbd-11eb-91bc-fe86699d79be.png)

![image](https://user-images.githubusercontent.com/83549410/116834240-6d8c1e00-abbd-11eb-89fa-61e83e3632c9.png)

All works like a charm. 12 threads seems to be just fine for now. (I hope RAM space is sufficient though).

**Comment by gntz555** [comment]
Yeah I'll be fine. Just 2 more plots. Amneamnius I thank you a lot. You probably not only helped me with this. I wish you the very best and stay healthy!

**Comment by gntz555** [comment]
![image](https://user-images.githubusercontent.com/83549410/116869707-7c042500-ac11-11eb-963a-debd720685c6.png)

Dunno why it only started 4 plots in generall. Also weirdly I got no notification for the finished plot, only the Plot Manager has started on ... message in Discord.

Edit: seems like I'm not the only one who has experienced it: https://github.com/swar/Swar-Chia-Plot-Manager/issues/53#issue-873789138

**Comment by gntz555** [comment]
I edited the config to 2 jobs and restarted the manager. Now it does more than 4 plots. 
![image](https://user-images.githubusercontent.com/83549410/116929381-a203e680-ac5e-11eb-9d24-ad5b753fe8e2.png)

**Comment by Amneamnius** [comment]
You had max for phase1 as 4 so that's why it didn't start more. After those finished up it should add more till it caps at max concurrent.

But if you have 2 temp drives then having 2 jobs like you have now is the way to go

**Comment by gntz555** [comment]
> You had max for phase1 as 4 so that's why it didn't start more. After those finished up it should add more till it caps at max concurrent.
> 
> But if you have 2 temp drives then having 2 jobs like you have now is the way to go

Yeah but that didn't change since yesterday. So I am puzzled. Anyways "micron" does 6 plots now. 4 plots are cruising on the slow HDD - as first directory (better than doing nothing for 7+ hours). It caps at 10 that's good. My bottleneck is the write/read speed. So I could add 32 threads to each and the CPU is still on 50% in average. Even my RAM isn't struggling.

So that are my settings for now.
[config.txt](https://github.com/swar/Swar-Chia-Plot-Manager/files/6417619/config.txt)
